### PR TITLE
chore(deps): update to Storybook 8

### PR DESCRIPTION
**Pull request by huwshimi**
## Done

- Upgrade Storybook deps to v8.
- Run migrations.
- Convert docs to CSF v3.
- Update docs to use autodoc.
- Fix iframe in prod. Fixes: #1102.

### Storybook

To see rendered examples of all react-components, run:

```shell
yarn start
```

### QA in your project

from `react-components` run:

```shell
yarn build
npm pack
```

Install the resulting tarball in your project with:

```shell
yarn add <path-to-tarball>
```

### QA steps

- Go through and check that component docs look OK.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -107,7 +107,7 @@
     "@types/react": "18.3.1",
     "@types/react-dom": "18.3.0",
     "postcss": "^8.3.11",
-    "jackspeak": "2.3.6"
+    "jackspeak": "2.1.1"
```
Comment (edlerd) [inline package.json:110]: Why reduce the version?

File: src/docs/getting-started.stories.mdx
Comment (edlerd) [inline src/docs/getting-started.stories.mdx:1]: Shall we keep this in some form?

**Review by edlerd — APPROVED** [review]
Spot checking QA on the demo server seems fine.
I didn't manage to check all the individual components. I guess those are mostly 1:1 transitions from mdx to tsx. Thanks for this massive update!

File: package.json (hunk 1 above)
Comment (huwshimi) [reply package.json:110]: The newer version doesn't work with Yarn 1.

File: src/docs/getting-started.stories.mdx
Comment (huwshimi) [reply src/docs/getting-started.stories.mdx:1]: Oops thanks, this file was renamed but the new file was not committed due to the .gitignore rules. I've fixed this now.

**Review by edlerd — CHANGES_REQUESTED** [review]
I am surprised all screenshots in percy are marked as "new"?
Also this one seems to be not all right:
image

**Comment by huwshimi** [comment]
> I am surprised all screenshots in percy are marked as "new"?

I think this is because the Storybook URL scheme has changed.

> Also this one seems to be not all right: 

Every time I run it a different set of images fail with that error. I'll try again later when hopefully the server is more stable.

**Comment by huwshimi** [comment]
I tried upgrading to the latest percy and storybook, but that didn't fix it.

**Comment by huwshimi** [comment]
@edlerd I've increased the asset loading timeouts in CircleCI and now the snapshots are all working.

File: .percy.yml [hunk 2]
```diff
@@ -2,3 +2,8 @@ version: 2
 snapshot:
   widths: [1280]
   enable-javascript: true
+discovery:
+  network-idle-timeout: 700
+  retry: true
+  launch-options:
+    timeout: 60000
```
Comment (edlerd) [inline .percy.yml:9]:
```suggestion
    timeout: 60000

```

**Review by edlerd — APPROVED** [review]
LGTM, tinny nitpick below.
It might be better to stash the commits into a single one?

File: .percy.yml (hunk 2 above)
Comment (huwshimi) [reply .percy.yml]: Thanks! Fixed.

**Pull request merged**